src/main/java/me/coley/recaf/search/SearchCollector.java

```diff
@@ -133,7 +132,7 @@ void addMatched(Context<?> context, Query quert) {
 		List<SearchResult> matched = quert.getMatched();
 		if(context != null)
 			matched.forEach(res -> res.setContext(context));
-		results.computeIfAbsent(quert, p -> new ArrayList<>()).addAll(matched);
+		results.computeIfAbsent(quert, p -> new ArrayList<>(matched.size())).addAll(matched);
```

Comment:
Why don't we take it a step further and just use [`Multimap`](https://guava.dev/releases/28.0-jre/api/docs/com/google/common/collect/Multimap.html) from Guava instead :wink:

```java
ListMultimap<Query, SearchResult> results = MultimapBuilder
		.linkedHashKeys(2).arrayListValues().build();
```

Oh, and there's a typo in the parameter `quert` by the way.